### PR TITLE
Update_richlist: Fix memory overflow when indexing addresses with 1000s of TXs

### PR DESCRIPTION
**Pull request by TheHolyRoger**
Richlist doesn't need to duplicate the TX list for every address indexed.

Also includes a function to delete entire richlist

### CODE REVIEW

**Comment by TheHolyRoger** [comment]
Some Sync script fixes added relating to this as well as a bit of whitespace cleanup that was annoying me :)

**Comment by TheHolyRoger** [comment]
@iquidus @cryptozeny would be nice if you could check this over :)

**Comment by cryptozeny** [comment]
i will check when i can. however i had no issue yet about this.

File: lib/database.js [hunk 1]
```diff
@@ -307,15 +307,15 @@ module.exports = {
   //property: 'received' or 'balance'
   update_richlist: function(list, cb){
     if(list == 'received') {
-      Address.find({}).sort({received: 'desc'}).limit(100).exec(function(err, addresses){
+      Address.find({}, 'a_id balance received').sort({received: 'desc'}).limit(100).exec(function(err, addresses){
```
Comment (TheHolyRoger) [inline lib/database.js]: This is the main fix to stop the memory overflows ^

File: lib/database.js [hunk 2]
```diff
@@ -307,15 +307,15 @@ module.exports = {
...
         Richlist.update({coin: settings.coin}, {
           received: addresses,
         }, function() {
           return cb();
         });
       });
     } else { //balance
-      Address.find({}).sort({balance: 'desc'}).limit(100).exec(function(err, addresses){
+      Address.find({}, 'a_id balance received').sort({balance: 'desc'}).limit(100).exec(function(err, addresses){
```
Comment (TheHolyRoger) [inline lib/database.js]: And this ^

**Pull request merged**